src/plugins/peer-view/public/js/peerview.js

```diff
@@ -252,14 +265,21 @@
           case 'mission-control-register':
             p.userName = msg[1];
             break;
+          case 'sys.ping':
+            p.sendemit('sys.pong', msg[1]);
+            break;
+          case 'mc-pong':
+            if (Number.isInteger(msg[1])){
+              p.lastping = Date.now() - msg[1];
+            }
```

Comment:
the new mc-pong message in the response sent from the webrtc peers to our mc-ping requests. It has the original host time that we sent the ping so we can easily calculate the latency.